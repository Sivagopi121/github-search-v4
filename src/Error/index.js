import React from 'react';
import styled from '@emotion/styled';

export const ErrorContainer = styled.div`
  margin: 20px;
  display: flex;
  justify-content: center;
`;

const ErrorMessage = ({ error }) => (
  <ErrorContainer>
    <small>{error.toString()}</small>
  </ErrorContainer>
);

export default ErrorMessage;
