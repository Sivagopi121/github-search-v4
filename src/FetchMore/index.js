import React from 'react';

import styled from '@emotion/styled';
import Loading from '../Loading';
import { ButtonUnobtrusive } from '../Button';

export const FetchMoreContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const FetchMore = ({
  loading,
  hasNextPage,
  variables,
  updateQuery,
  fetchMore,
  children,
}) => (
  <FetchMoreContainer>
    {loading ? (
      <Loading />
    ) : (
      hasNextPage && (
        <ButtonUnobtrusive
          style={{
            margin: '20px 0',
          }}
          onClick={() => fetchMore({ variables, updateQuery })}
        >
          More
          {' '}
          {children}
        </ButtonUnobtrusive>
      )
    )}
  </FetchMoreContainer>
);

export default FetchMore;
