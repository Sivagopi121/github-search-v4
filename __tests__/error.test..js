import React from 'react';
import renderer from 'react-test-renderer';
import Error from '../src/Error/index'
import FetchMore from '../src/FetchMore';
import Input from '../src/Input/index'
import Loading from '../src/Loading';

test('Error renders correctly', () => {
  const tree = renderer
    .create(
      <Error error="Some error"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


test('Input renders correctly', () => {
  const tree = renderer
    .create(
      <Input/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('FetchMore renders correctly', () => {
  const tree = renderer
    .create(
      <FetchMore/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


test('Loading renders correctly', () => {
  const tree = renderer
    .create(
      <Loading/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});