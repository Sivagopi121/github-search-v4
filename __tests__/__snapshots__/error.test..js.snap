// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<div
  className="css-1dhx6mc-ErrorContainer e1lis3cs0"
>
  <small>
    Some error
  </small>
</div>
`;

exports[`FetchMore renders correctly 1`] = `
<div
  className="css-wn5agb-FetchMoreContainer e5zzf080"
/>
`;

exports[`Input renders correctly 1`] = `
<input
  className="Input Input_black"
/>
`;

exports[`Loading renders correctly 1`] = `
<div
  className="Loading"
>
  <small>
    Loading
     
  </small>
</div>
`;
